pkg/nfd-gc/nfd-gc.go

```diff
@@ -190,7 +198,27 @@ func (n *nfdGarbageCollector) garbageCollect() {
 
 	// Handle NodeResourceTopology objects
 	listAndHandle(gvrNRT, func(meta metav1.PartialObjectMetadata) {
-		if !nodeNames.Has(meta.Name) {
+		deleteNRT := false
+
+		if label, ok := meta.Annotations[nfdtopologyupdater.NRTOwnerPodAnnotation]; ok {
+			if s := strings.Split(label, "_"); len(s) == 2 {
+				ns := s[0]
+				pod := s[1]
+				_, err := n.k8sClient.CoreV1().Pods(ns).Get(context.TODO(), pod, metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					if _, ok := gcNRTs[meta.Name]; !ok {
+						gcNRTs[meta.Name] = label
+					} else {
```

Comment:
Wouldn't we need to check the value as well? `val == label`. If the pod name (or namespace) has changed, DON'T delete the NRT.

I just realized that the `label` is somewhat misleading name for the variable. How about e.g. `objRef` or `objKey` or smth?